### PR TITLE
br: explicitly set analyze sample number if tidb_analyze_version = 2

### LINKED ISSUE #27391: lightning analyze table parameter change to avoid tidb OOM
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

When tidb global.tidb_analyze_version=2 and lightning import 70GB table in TiDB, TiDB might OOM during analyze phase of lightning import.
./tidb-lightning -config /minio/fubin/dbgen/v2data/lightning.toml

### 2. What did you expect to see? (Required)
lightning import succeed, and TiDB not OOM

### 3. What did you see instead (Required)
TiDB OOM

### 4. What is your TiDB version? (Required)
TiDB: 5.2.0
Lightning: 5.2.0

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #27391

Problem Summary: Analyze table OOM easily when `tidb_analyze_version = 2`

### What is changed and how it works?
Add explicitly analyze parameter `WITH NUM 10000` if the TiDB's `tidb_analyze_version = 2` and there are more than 600 regions in target table.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - test with 100GB source data, before this pr, target tidb(8core, 16GB mem) always OOM; after this PR, analyze run successfully with tidb peak memory about 1.5GB. BTW, run `analyze table` and `analyze table with 10000 samples` results in the same result(with `tidb_build_stats_concurrency = 20;` which the default lightning config).
- [ ] No code

Side effects

Documentation

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by kennytm** [comment]
release-5.2? is this a cherry-pick of some PR not listed, or #27391 does not affect master?

**Comment by glorv** [comment]
> release-5.2? is this a cherry-pick of some PR not listed, or #27391 does not affect master?

@kennytm  This is only a work-around for v5.2 due to the unmature of v2 analyze.

cc @chrysan @winoros

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: br/pkg/lightning/restore/table_restore.go [hunk 1]
```diff
@@ -886,9 +889,49 @@ func (tr *TableRestore) compareChecksum(remoteChecksum *RemoteChecksum, localChe
 	return nil
 }
 
+func (tr *TableRestore) genAnalyzeSQL(ctx context.Context, g glue.SQLExecutor) string {
+	// TODO: because tidb v5.2 analyze full table OOM easily, we fallback to add `WITH NUM 10000` parameter
+	// if there are more than 600 regions in this table
+	sql := "ANALYZE TABLE "+tr.tableName
+	result, err := g.QueryStringsWithLog(ctx, "SHOW VARIABLES LIKE 'tidb_analyze_version'", "fetch tidb analyze version", tr.logger)
```
Comment (kennytm) [inline br/pkg/lightning/restore/table_restore.go:896]: why not just `select @@tidb_analyze_version` 😕

Comment (glorv) [reply br/pkg/lightning/restore/table_restore.go:896]: I originally want to be compatible with v4.0, but It's simplier to fallback to 1 on error.

**Review by kennytm — APPROVED** [review]

**Pull request merged**